ui/modal/controller.go

```diff
@@ -75,13 +83,46 @@ func (m ViewModel) HandleMessage(msg tea.Msg) (*ViewModel, tea.Cmd) {
 			// If the previous state is not active
 			if ok {
 				if !m.transactionModal.Active {
-					if acct.Participation != nil &&
-						acct.Participation.VoteFirstValid == m.transactionModal.Participation.Key.VoteFirstValid {
-						m.SetActive(true)
-						m.infoModal.Active = true
-						m.infoModal.Prefix = "Successfully registered online!\n"
-						m.HasPrefix = true
-						m.SetType(app.InfoModal)
+					if acct.Participation != nil {
+						// comparing values to detect corrupted/non-resident keys
+						fvMatch := boolToInt(acct.Participation.VoteFirstValid == m.transactionModal.Participation.Key.VoteFirstValid)
```

Comment:
This could just be the value of the boolean and reused below for the checks. No need to recompute this with an equality check

Even more preferable would be a `participation.IsEqual(a, b)` interface but it would need a custom error object in order to receive the invalid keys. Not necessary but would be more robust